pandas/_libs/lib.pyx

```diff
@@ -1742,6 +1753,21 @@ cpdef bint is_datetime_array(ndarray values):
     return validator.validate(values)
 
 
+cdef class DatetimeTZValidator(TemporalValidator):
+    cdef bint is_value_typed(self, object value) except -1:
+        return PyDateTime_Check(value) and (value.tzinfo is not None)
+
+    cdef inline bint is_valid_null(self, object value) except -1:
+        return is_null_datetime64(value)
```

Comment:
we might want to only accept `value is NaT` here

Reply:
@jbrockmendel sorry but I think that we should at least test for None as well since infer_dtype will be used on arrays we receive from the various SQL drivers and None is AFAIK the default null value there. As you can see in the example below (this is the issue that we are currently trying to fix) the array has not been parsed to a DatetimeTZDtype and still contains None:

```python
import pandas as pd
from sqlalchemy import create_engine

# REPLACE with valid postgresql connection string
engine = create_engine('postgresql://............')

df = pd.DataFrame({'dt': [pd.Timestamp('2019-11-14 15:12:00+0000', tz='UTC'),
                          pd.Timestamp('2019-08-07 13:37:04+0000', tz='UTC'),
                          pd.NaT]})
df.to_sql('null_value_test', con=engine, schema='tests', if_exists='replace',
          index=False)
pd.read_sql('SELECT * FROM tests.null_value_test', con=engine)
```
|    | dt                        |
|---:|:--------------------------|
|  0 | 2019-11-14 16:12:00+01:00 |
|  1 | 2019-08-07 15:37:04+02:00 |
|  2 |                      None |

Reply:
Thank you for clarifying.  I'll need to give some thought to whether this is the appropriate place to do this check.